src/scripts/ui/ResourceImportComponent.tsx

```diff
@@ -33,6 +33,8 @@ import { showModal } from "./GlobalModal";
 import { FormatNumber } from "./HelperComponents";
 import { TableView } from "./TableView";
 
+const resourceImportSortingState = { column: 0, asc: true };
```

Comment:
This is perhaps wrong? `column 0` is not sortable

Reply:
Yes you are right.

Reply:
TableView API should probably validate that the sorting state passed in is valid. And log an error if it isn't

Reply:
Probably a good idea. I will take a look at that tomorrow.

Reply:
Would something like this do?

```let defaultSortingColumn: number = header.findIndex((v) => v.sortable);
   if (header.at(sortingState?.column!)?.sortable) {
      defaultSortingColumn = sortingState?.column!;
   }
   const [sortColumn, setSortColumn] = useState(defaultSortingColumn);

Reply:
Or simply `console.assert(header[sortColumn].sortable)`?

Reply:
I would have thought checking if the sorting state column was valid and if not falling back to the first one that is.

I have put the assert in though.

Reply:
Since this is not a user input, but rather an functional call, it's okay to use assertions to indicate to the caller that the assumptions the function make